.circleci/ci/lighthouse.js

```diff
@@ -0,0 +1,143 @@
+        writeObjLighthouseConfig(runnerResult.lhr.categories, lighthouseConfig)
+    }
+    await chrome.kill();
+}
+
+const getCommentText = (resultCategories) => {
+  const commentText = `### Lighthouse scores\n\n|        | Performance | Accessibility | Best-Practices | SEO |\n| ------ | ----------- | ------------- | -------------- | --- |\n| Scores |     ${resultCategories.performance.score*100}      |       ${resultCategories.accessibility.score*100}       |       ${resultCategories['best-practices'].score*100}       |  ${resultCategories.seo.score*100} |`
+  return commentText
+}
+
+const checkThresholds = (resultCategories, lighthouseConfig) => {
+    const {  performance, accessibility, bestPractices, seo } = lighthouseConfig.requiredScores[0]
+    const margin = lighthouseConfig.margin
+    const thresholdResults = {
+        isThresholdPass: false,
+        updateLighthouseConfig: false
+    }
+    if(performance*(1-margin) < resultCategories.performance.score*100 &&
+        accessibility*(1-margin) < resultCategories.accessibility.score*100 &&
+        bestPractices*(1-margin) < resultCategories['best-practices'].score*100 &&
+        seo*(1-margin) < resultCategories.seo.score*100){
+            thresholdResults.isThresholdPass = true
+        }
+
+    if(performance < resultCategories.performance.score*100 &&
+        accessibility > resultCategories.accessibility.score*100 &&
+        bestPractices > resultCategories['best-practices'].score*100 &&
+        seo > resultCategories.seo.score*100){
+            thresholdResults.updateLighthouseConfig = true
+        }
+
+        return thresholdResults
+
+}
+
+const postCommentToGitHub = (commentText, GITHUB_TOKEN) => {
+
+    const {CIRCLE_PROJECT_USERNAME, CIRCLE_PROJECT_REPONAME, CIRCLE_PULL_REQUEST } = process.env
+    const prNumber = CIRCLE_PULL_REQUEST.split('/').pop();
+    const apiUrl = new URL(`https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/issues/${prNumber}/comments`);
+    const postData = JSON.stringify({body: commentText});
+
+    // Define the options for the HTTPS request
+    const options = {
+      method: 'POST',
+      headers: {
+        'Authorization': `${GITHUB_TOKEN}`,
+        'User-Agent': 'CircleCI'
+      }
+    };
+
+    // Send the HTTPS request to create the new comment on the pull request
+    const req = https.request(apiUrl, options, (res) => {
+      console.log(`Status: ${res.statusCode}`);
+      res.on('data', (d) => {
+        process.stdout.write(d);
+      });
+    });
+
+    req.on('error', (error) => {
+      console.error(error);
+    });
+
+    req.write(postData);
+    req.end();
+}
+
+const writeObjLighthouseConfig = (resultCategories, lighthouseConfig) => {
+
+const {performanceResult, accessibilityResult, bestPracticesResult, seoResult} = {performanceResult: resultCategories.performance.score, accessibility: resultCategories.accessibility.score,
```

Comment:
indentation and code readability is poor. pls improve.

Reply:
done..